tests/phpunit/tests/webfonts-api/wpWebfontsSchemaValidator.php

```diff
@@ -0,0 +1,470 @@
+				'webfont'          => array(
+					'provider'    => null,
+					'font-family' => 'Some Font',
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+				),
+				'expected_message' => 'Webfont provider must be a non-empty string.',
+			),
+			'font-family: not defined'            => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+				),
+				'expected_message' => 'Webfont font family must be a non-empty string.',
+			),
+			'font-family: empty string'           => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => '',
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+				),
+				'expected_message' => 'Webfont font family must be a non-empty string.',
+			),
+			'font-family: not a string'           => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => null,
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+				),
+				'expected_message' => 'Webfont font family must be a non-empty string.',
+			),
+			'src: not defined'                    => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => 'Source Serif Pro',
+					'font-style'  => 'normal',
+					'font-weight' => '200 900',
+				),
+				'expected_message' => 'Webfont src must be a non-empty string or an array of strings.',
+			),
+			'src: type is invalid'                => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => 'Source Serif Pro',
+					'font-style'  => 'normal',
+					'font-weight' => '200 900',
+					'src'         => null,
+				),
+				'expected_message' => 'Webfont src must be a non-empty string or an array of strings.',
+			),
+			'src: individual src is not a string' => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => 'Source Serif Pro',
+					'font-style'  => 'normal',
+					'font-weight' => '200 900',
+					'src'         => array( null ),
+				),
+				'expected_message' => 'Each webfont src must be a non-empty string.',
+			),
+			'src: invalid url'                    => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => 'Source Serif Pro',
+					'font-style'  => 'normal',
+					'font-weight' => '200 900',
+					'src'         => '/assets/fonts/font.woff2',
+				),
+				'expected_message' => 'Webfont src must be a valid URL or a data URI.',
+			),
+			'src: invalid data uri'               => array(
+				'webfont'          => array(
+					'provider'    => 'local',
+					'font-family' => 'Source Serif Pro',
+					'font-style'  => 'normal',
+					'font-weight' => '200 900',
+					'src'         => 'data:text/plain',
+				),
+				'expected_message' => 'Webfont src must be a valid URL or a data URI.',
+			),
+		);
+	}
+
+	/**
+	 * @covers       WP_Webfonts_Registry::set_valid_properties
```

Comment:
This covers WP_Webfonts_Schema_Validator, same in the other methods.

```suggestion
	 * @covers       WP_Webfonts_Schema_Validator::set_valid_properties
```

Reply:
Great catch @Soean! Fixed in https://github.com/WordPress/wordpress-develop/pull/1736/commits/7939995d89f495560d29945b576292168084539f 👍